CMakeLists.txt

```diff
@@ -0,0 +1,109 @@
+
+  # Get the toolchain variant
+  # TODO: Add support for cross-compile toolchain variant
+  # TODO: Enforce GCC version check against TF-M compiler requirements
+  if(${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "zephyr")
+    set(TFM_TOOLCHAIN "GNUARM")
+    set(TFM_TOOLCHAIN_PREFIX "arm-zephyr-eabi")
+    set(TFM_TOOLCHAIN_PATH ${ZEPHYR_SDK_INSTALL_DIR}/arm-zephyr-eabi)
+  elseif(${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "gnuarmemb")
+    set(TFM_TOOLCHAIN "GNUARM")
+    set(TFM_TOOLCHAIN_PREFIX "arm-none-eabi")
+    set(TFM_TOOLCHAIN_PATH ${GNUARMEMB_TOOLCHAIN_PATH})
+  else()
+    message(FATAL_ERROR "Unsupported ZEPHYR_TOOLCHAIN_VARIANT: ${ZEPHYR_TOOLCHAIN_VARIANT}")
+  endif()
+
+  include(ExternalProject)
+
+  ExternalProject_Add(
+    tfm
+    SOURCE_DIR ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m
+    BINARY_DIR ${TFM_BINARY_DIR}
+    CMAKE_ARGS -DPROJ_CONFIG=${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/configs/${TFM_CFGFILE}.cmake
+               -DTARGET_PLATFORM=${TFM_BOARD}
+               -DBL2=True
+               -DCOMPILER=${TFM_TOOLCHAIN}
+               -DGNUARM_PREFIX=${TFM_TOOLCHAIN_PREFIX}
+               -DGNUARM_PATH=${TFM_TOOLCHAIN_PATH}
+    BUILD_ALWAYS True
+    USES_TERMINAL_BUILD True
+    BUILD_BYPRODUCTS ${VENEERS_FILE}
+  )
+
+  set(TFM_INTERFACE_SOURCE ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/src)
+
+  # IPC mode source dependencies
+  if(TFM_IPC)
+    add_library(tfm_ipc_psa_api
+                ${TFM_INTERFACE_SOURCE}/tfm_psa_ns_api.c
+                ${TFM_INTERFACE_SOURCE}/tfm_crypto_ipc_api.c
+                ${TFM_INTERFACE_SOURCE}/tfm_initial_attestation_ipc_api.c
+                ${TFM_INTERFACE_SOURCE}/tfm_its_ipc_api.c
+                ${TFM_INTERFACE_SOURCE}/tfm_platform_ipc_api.c
+                ${TFM_INTERFACE_SOURCE}/tfm_sst_ipc_api.c
+                ${TFM_INTERFACE_SOURCE}/tfm_ns_interface.c
+    )
+    target_include_directories(tfm_ipc_psa_api PUBLIC ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/include)
+    target_compile_definitions(tfm_ipc_psa_api PUBLIC TFM_PSA_API TFM_PARTITION_TEST_CORE_IPC)
+    target_link_libraries(tfm_ipc_psa_api PRIVATE zephyr_interface)
+  endif()
+
+  # Library mode source dependencies
+  if(TFM_LIB)
+   add_library(tfm_func_api
+               ${TFM_INTERFACE_SOURCE}/tfm_ns_interface.c
+               ${TFM_INTERFACE_SOURCE}/tfm_nspm_api.c
+               ${TFM_INTERFACE_SOURCE}/tfm_nspm_svc_handler.c
+               ${TFM_INTERFACE_SOURCE}/tfm_audit_func_api.c
+               ${TFM_INTERFACE_SOURCE}/tfm_crypto_func_api.c
+               ${TFM_INTERFACE_SOURCE}/tfm_initial_attestation_func_api.c
+               ${TFM_INTERFACE_SOURCE}/tfm_its_func_api.c
+               ${TFM_INTERFACE_SOURCE}/tfm_platform_func_api.c
+               ${TFM_INTERFACE_SOURCE}/tfm_sst_func_api.c
+   )
+   target_include_directories(tfm_func_api PUBLIC ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/include)
```

Comment:
I suspect this line is needed here:
```
target_link_libraries(tfm_ipc_psa_api PRIVATE zephyr_interface)
```
have you tried building with library mode ?